src/assert.js

```diff
@@ -305,6 +305,100 @@ class Assert {
 			message
 		} );
 	}
+
+	rejects( promise, expected, message ) {
+		let result = false;
+
+		const currentTest = ( this instanceof Assert && this.test ) || config.current;
+
+		// 'expected' is optional unless doing string comparison
+		if ( objectType( expected ) === "string" ) {
+			if ( message == null ) {
+				message = expected;
+				expected = null;
+			} else {
+				throw new Error(
```

Comment:
I may've missed something here, but it seems reasonable to support `expected` as a string, especially given it is documented as being a comparable for the rejection value. The rejection value can be a string, the first documentation example added in this commit also suggests it:

```
Promise.reject("error")
```

Based on the documentation, I would expect that argument types Function and Regex are special-cased, and that other types (e.g. Error instance) are handled by something like propEqual somehow, given how it compares the `name` and `message` properties. I realise the implementation below doesn't use propEqual, and I'm not suggesting it does, but this is what it looks like from the outside, and I'd be nice if string, number and boolean rejection values would work. Alternatively, we'd need to stronger document that the only comparable rejection value we support is an object, and we should also strictly document in that case that that case too is a special case, unlike propEqual or deepEqual, which specifically compares the `constructor`, `name` and `message` properties only.